src/validation/data.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2023 Helsing GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{
+    collections::{btree_map::Entry, BTreeMap},
+    path::PathBuf,
+};
+
+use diff::Diff;
+use miette::Diagnostic;
+use protobuf::descriptor::{
+    field_descriptor_proto::{Label as FieldDescriptorLabel, Type as FieldDescriptorType},
+    *,
+};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use crate::validation::{
+    rules::{Rule, RuleSet},
+    Violations,
+};
+
+mod entity;
+mod r#enum;
+mod message;
+mod package;
+mod packages;
+mod service;
+
```

Comment:
If this is absolutely required to use the validation mod, i would call this `prelude` to indicate that this glob-preludes types for you

Reply:
Thats correct, im more talking about the layer above.

So the data mod seems _weird_ to me in the sense that it pollutes the pub api a lot and im trying to see if there is a possibility to condense this.

I would not offer any in-depth public api currently (even if we are still unstable). 

So in my opinion the pub api should be condensable to:

```
// <- on the package type, not the store type imo!
impl Package {
    pub fn validate(&self) -> miette::Result<Vec<Violations>> { /* .. */ }
}

// and maybe one standalone one:
pub fn validate(protos: impl AsRef<[std::file::File]>, manifest: impl AsRef<Manifest>) -> miette::Result<..> {}
```

---

Everything else should be hidden inside of the validations module. This allows us to "mess" with the internals without people starting to use it in buffrs, the registry or even worse externally. Also as long as there is no need for custom rules (we will notice, once there is!) this doesnt hurt at all imo

Reply:
What is the status? 👀

Reply:
Everything is private now.